src/main.rs

```diff
@@ -40,14 +33,31 @@ async fn main() {
     });
```

Comment:
The initial model update should happen before the application loads. This can be changed so that the control flow is sequential.